src/main/java/com/cleanroommc/groovyscript/sandbox/MixinSandbox.java

```diff
@@ -0,0 +1,256 @@
+            throw new RuntimeException(e);
+        }
+        // inject loaded mixin classes into configuration
+        mixinClasses.addAll(groovyMixins);
+        instance.getEngine().writeIndex();
+    }
+
+    private static MixinSandbox instance;
+    private static final Map<String, byte[]> resourceCache;
+
+    static {
+        Map<String, byte[]> resourceCache1;
+        try {
+            Field field = LaunchClassLoader.class.getDeclaredField("resourceCache");
+            field.setAccessible(true);
+            resourceCache1 = (Map<String, byte[]>) field.get(Launch.classLoader);
+        } catch (NoSuchFieldException | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+        resourceCache = resourceCache1;
+    }
+
+    public static final Logger LOG = LogManager.getLogger("GroovyScript-MixinSandbox");
+    public static final boolean DEBUG = true;
+
+    private MixinSandbox() {}
+
+    @Override
+    protected ScriptEngine createEngine(CompilerConfiguration config) {
+        return new ScriptEngine(SandboxData.getRootUrls(), SandboxData.getMixinScriptCachePath(), SandboxData.getScriptFile(), config)
+                .onClassLoaded(cc -> {
+                    GroovyLog.get().info(" - loaded mixin class {} from cache", cc.getName());
+                    resourceCache.put(cc.getName(), cc.getData());
+                });
+    }
+
+    @Override
+    protected void initConfig(CompilerConfiguration config) {
+        getImportCustomizer().addImports(
+                Arrays.asList(
+                        Mixin.class,
+                        Inject.class,
+                        At.class,
+                        CallbackInfo.class,
+                        CallbackInfoReturnable.class,
+                        Coerce.class,
+                        Constant.class,
+                        Desc.class,
+                        Descriptors.class,
+                        Group.class,
+                        ModifyArg.class,
+                        ModifyArgs.class,
+                        ModifyConstant.class,
+                        ModifyVariable.class,
+                        Redirect.class,
+                        Slice.class,
+                        Surrogate.class,
+                        Final.class,
+                        Mutable.class,
+                        Overwrite.class,
+                        Pseudo.class,
+                        Shadow.class,
+                        Unique.class,
+                        SoftOverride.class,
+                        Implements.class,
+                        Interface.class,
+                        Intrinsic.class,
+                        WrapOperation.class,
+                        ModifyExpressionValue.class,
+                        ModifyReceiver.class,
+                        ModifyReturnValue.class,
+                        Local.class,
+                        Share.class)
```

Comment:
to accompany `Local` and `Share` i would suggest `LocalRef` and its primitive variants, along with `Cancellable`. theres also `WrapMethod`. 
there is also `Invoker` and `Accessor` - given the native groovy features, i dont believe these are needed?
it might be that the version of mixinbooter in the dev env is older than some of these - recently updated the buildscript repo to use a newer version of mixinbooter, so i would suggest updating the buildscript.